api/app/services/ingestions/strategies/abstract_strategy.rb

```diff
@@ -0,0 +1,55 @@
+module Ingestions
+  module Strategies
+    # @abstract
+    class AbstractStrategy < Ingestions::IngestionService
+      define_model_callbacks :ingestibility_check, :perform
+
+      boolean :test_only, default: false
+
+      set_callback :perform, :before, -> { context.set_used_strategy! strategy_name }
+
+      def execute
+        run_callbacks :ingestibility_check do
+          @ingestible = determine_ingestibility
+        end
+
+        return ingestible? if test_only
+        return nil unless ingestible?
+
+        run_callbacks :perform do
+          @manifest = perform
+        end
+
+        manifest
+      end
+
+      def ingestible?
+        @ingestible
+      end
+
+      attr_reader :manifest
+
+      # @abstract
+      # @return [Boolean]
+      def determine_ingestibility
+        raise NotImplementedError, "Must implement #{self.class}##{__method__}"
+      end
+
+      # @return [Manifest]
+      def perform
+        raise NotImplementedError, "Must implement #{self.class}##{__method__}"
+      end
+
+      def inspector
+        @inspector ||=
+          "Ingestions::Strategy::#{strategy_name}".safe_constantize::Inspector.new context
```

Comment:
You want to be using `constantize` here, not `safe_constantize`. If this class does not exist we want a `NameError` to be raised.

Also, when dealing with abstract classes like this, since this is something that may get overridden by users, we want to provide this as a hook.

```ruby
# @api private
# @abstract
# @return [Class]
def inspector_klass
  "Ingestions::Strategy::#{strategy_name}::Inspector".constantize
end

def inspector
  @inspector ||= inspector_klass.new context
end
```